### PR TITLE
UserContribution

### PR DESCRIPTION
**Pull request by RaimondSob**
Adding bars and adding drinks now works through app

### CODE REVIEW

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/AddBarPage.xaml.cs [hunk 1]
```diff
@@ -5,15 +5,63 @@
...
+            
+            Bar bar = new Bar(barName.Text,barAdress.Text);
+
+            var restService = new RestService();
+            BarList = Task.Run(async () => await restService.GetData<List<Bar>>(Constants.BarsUrl)).Result;
+
+            foreach (Bar searchBar in BarList)
+            {
+                
+                if (searchBar.BarName == bar.BarName && searchBar.BarLocation == bar.BarLocation)
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/AddBarPage.xaml.cs:56]: searchBar might conflict later on with an actual searchbar

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/DeleteDrinkPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        public DeleteDrinkPage()
+        {
+            InitializeComponent();
+            this.BindingContext = new DeleteDrinkViewModel();
+        }
+
+        private void Button_Clicked(object sender, EventArgs e)
+        {
+
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/DeleteDrinkPage.xaml.cs:23]: unnecessary whitespace

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs [hunk 3]
```diff
@@ -6,16 +6,101 @@
...
             InitializeComponent();
             this.BindingContext = new EditDrinkPriceViewModel();
+            var restService = new RestService();
+            initialBarList = Task.Run(async () => await restService.GetData<List<Bar>>(Constants.BarsUrl)).Result;
+            initialDrinkList = Task.Run(async () => await restService.GetData<List<Drink>>(Constants.DrinksUrl)).Result;
+        }
+
+        private async void OnSaveClicked(object sender, EventArgs e)
+        {
+
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs:63]: unnecessary whitespace

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs [hunk 4]
```diff
@@ -6,16 +6,101 @@
...
+            if (Regex.IsMatch(newEntryPrice.Text, @"(^[1-9]\d*(.\d{1,2})?$)|(^0(\.\d{1,2})?$)"))
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs:64]: regex probably should have been moved to a different class

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs [hunk 5]
```diff
@@ -6,16 +6,101 @@
...
+                SpecificPrice newPrice = new SpecificPrice
+                {
+                    BarID = bar.BarID,
+                    DrinkID = drink.DrinkID,
+                    DrinkPrice = price
+                };
+                var restService = new RestService();
+                await restService.SaveData<SpecificPrice>(newPrice,Constants.SpecificPricesUrl, false);
+            }
+
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs:86]: whitespace

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs [hunk 6]
```diff
@@ -6,16 +6,101 @@
...
+            else 
+            {
+                await DisplayAlert("Warning", "Bad Input", "Ok");
+            }
+
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs:91]: whitespace

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs [hunk 7]
```diff
@@ -6,16 +6,101 @@
...
+        private void barChoice_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            List<Drink> tempDrinkList = new List<Drink>();
+            
+            foreach (var SpecificPrice in ((Bar)barChoice.SelectedItem).AvailableDrinks)
+            {
+                tempDrinkList.Add(initialDrinkList.Find(x => x.DrinkID == SpecificPrice.DrinkID));
+            }
+
+            DrinkList = tempDrinkList;
```
Comment (ElektrosStulpas) [inline WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs:103]: this maybe could have been done with async

**Review by ElektrosStulpas — COMMENTED** [review]
that's a lot of code, worried only about the public and private as same variable and in some places seems like it should be related to Drink instead of bar

File: WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/AddBarViewModel.cs [hunk 8]
```diff
@@ -1,10 +1,12 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text;
+using Xamarin.Forms;
 
 namespace ComparisonEngineUI.ViewModels
 {
     class AddBarViewModel : BaseViewModel
     {
+        //public Command SaveCommand { get; };
```
[the remark this answers is not in the document]
Comment (RaimondSob) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/AddBarViewModel.cs:10]: I think later we will. We need to take out the saving logic out of xaml.cs files to viewmodels, that would be the proper way, but don't really know how to do that, but we'll probably use Command instead of Clicked, so it makes sense to leave it for now

File: WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/AddDrinkViewModel.cs [hunk 9]
```diff
@@ -1,10 +1,45 @@
...
+        {
+            
+           // SaveCommand = new Command(OnSaveClicked);
+            var restService = new RestService();
+            BarList = Task.Run(async () => await restService.GetData<List<Bar>>(Constants.BarsUrl)).Result;
+            
+        }
+        private async void OnSaveClicked(object obj)
+        {
+
```
[the remark this answers is not in the document]
Comment (RaimondSob) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/AddDrinkViewModel.cs:42]: Same as in previous, this is for future(to have logic in viewmodels, not in views)

File: WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/DeleteDrinkViewModel.cs [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+                }
+            }
+        }
+        public DeleteDrinkViewModel()
+        {
+            var restService = new RestService();
+            BarList = Task.Run(async () => await restService.GetData<List<Bar>>(Constants.BarsUrl)).Result;
+            
+        }
+    }
```
[the remark this answers is not in the document]
Comment (RaimondSob) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/DeleteDrinkViewModel.cs:35]: As I see it, it should have both, user picks from which bar he wants to delete, then he picks which drinks in that bar he wants to delete(It would be weird if user just choose only one drinks, which would be deleted from all bars). The problem with this is having list of drinks corresponding to which bar was picked.

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/AddBarPage.xaml.cs (hunk 1 above)
Comment (RaimondSob) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/Views/AddBarPage.xaml.cs:56]: I agree. Actually, a lot of names given should be renamed for better readability of the code, will do that later.

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs (hunk 4 above)
Comment (RaimondSob) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs:64]: To different class maybe, but not further(I think we were discussing about moving regex to backend), as I understand regex, it's more of a front end thing, because it doesn't allow me to enter anything but it's set format, so having that in backend could probably slow down the entering.

File: WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs (hunk 7 above)
Comment (RaimondSob) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/Views/EditDrinkPricePage.xaml.cs:103]: Yes, that does make sense, but this part of code is actually not doing anything at this moment.

File: WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/AddDrinkViewModel.cs [hunk 11]
```diff
@@ -1,10 +1,45 @@
...
+                    OnPropertyChanged();
+                }
+            }
+        }
+        public AddDrinkViewModel()
+        {
+            
+           // SaveCommand = new Command(OnSaveClicked);
+            var restService = new RestService();
+            BarList = Task.Run(async () => await restService.GetData<List<Bar>>(Constants.BarsUrl)).Result;
```
[the remark this answers is not in the document]
Comment (RaimondSob) [reply WebService/ComparisonEngineUI/ComparisonEngineUI/ViewModels/AddDrinkViewModel.cs:37]: User can't just add drink, he first needs to choose to which bar he wants to add drink.
